File: 2020/Python/01.py
import itertools

def generator(input) :
    return [int(expense) for expense in input.splitlines()]

def part_1(input) :
    for (a,b) in list(itertools.product(input, input)):
        if a + b == 2020 : 
            return a * b

def part_2(input) : 
    for (a,b, c) in list(itertools.product(input, input, input)):
        if a + b + c == 2020 : 
            return a * b * c

File: 2017/Rust/src/bin/01.rs
aoc2017::main!();

fn generator(input : &str) -> Vec<u32> {
    input
        .chars()
        .map(|digit| digit.to_digit(10).unwrap()) 
        .collect_vec()
}

fn part_1(input : Vec<u32>) -> u32 {
    solver(input, 1)

}

fn part_2(input : Vec<u32>) -> u32 {
    solver(input.clone(), input.len()/2)
}

fn solver(input : Vec<u32>, shift : usize) -> u32 {
    input
    .iter()
    .enumerate()
    .filter(|(pos, digit)| *digit == input.iter().nth((pos + shift) % input.len()).unwrap())
    .fold(0, |acc, digit| acc + digit.1)
}